### ISSUE TITLE
`v-select` doesn't update selected item, when `items` array doesn't contain selected value initially, but is added later

### ISSUE THREAD

**Issue by prakashgp** [vuetifyjs/vuetify#1738]
<v-select label="Country" autocomplete  required :items="items"
						
						v-model="selected"
						item-text="name" item-value="_id"
						>
						</v-select>
export default {
data: () => ({
selected: 'value1',
items: []
}),
mounted() {
setTimeout(function() {
this.items.push({_id: 'value1', name: 'Selected item'}); // the items's value is the selected value, but doesn't reflect in 'v-select' after adding
}, 100);
}
}

**Comment by jacekkarczmarczyk** [comment]
That's interesting case, but could you please use the bug report template and provide codepen link for this issue (also in the second one)? It will help the dev team a lot, thank you!

**Comment by jacekkarczmarczyk** [comment]
Test proposal

```js
  it('should not set the value that is not in the items list', async () => {
    const wrapper = mount(VSelect, {
      propsData: {
        value: 'invalid',
        items: []
      }
    })

    await wrapper.vm.$nextTick()
    expect(wrapper.vm.value == null).toBe(true)

    wrapper.setProps({items: ['invalid']})
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.value == null).toBe(true)

    expect('Application is missing <v-app> component.').toHaveBeenTipped()
  })
```

**Comment by jacekkarczmarczyk** [comment]
### Test proposal

```js
  it('should not set the value that is not in the items list', async () => {
    const wrapper = mount(VSelect, {
      propsData: {
        value: 'invalid',
        items: ['valid']
      }
    })

    await wrapper.vm.$nextTick()
    expect(wrapper.vm.inputValue == null).toBe(true)

    expect('Application is missing <v-app> component.').toHaveBeenTipped()
  })

  it('should not set the value that is not in the items list (return object)', async () => {
    const wrapper = mount(VSelect, {
      propsData: {
        value: 'invalid',
        returnObject: true,
        itemText: 'text',
        itemValue: 'value',
        items: [{text: '', value: 'valid'}]
      }
    })

    await wrapper.vm.$nextTick()
    expect(wrapper.vm.inputValue == null).toBe(true)

    expect('Application is missing <v-app> component.').toHaveBeenTipped()
  })
```

**Comment by prakashgp** [comment]
but it would be better if it does so in case of ajax autocomplete it would
be very helpful.  please check vue multiselect library that does so

**Comment by nekosaur** [comment]
Please update your issue using the provided template for issues, including a reproduction link, or the issue will be closed.

**Comment by leandrobortolotto** [comment]
I had this issue too, and here it is how I solved it:
`
                <v-select
                  name="marketSegment"
                  label="Segmento"
                  id="marketSegment"                
                  v-bind:items="marketSegments"
                  v-model="local.marketSegments"
                  multiple
                  item-text="name"
                  item-value="marketSegmentId"
                  return-object
                  autocomplete
                ></v-select>
`
The problem occurs because of async data, I change how I load the data and it worked.

Bottom line, I had to load marketSegments variable (which has all possible values) first and then load the previously selected data, in this case, `this.local.marketSegments`

Final solution:
`
  created: function () {
    // this.fetchMarketSegments()
    var self = this
    this.$store.dispatch('fetchMarketSegments')
    .then((response) => {
      self.marketSegments = response.data
      self.local = self.$store.getters.fetchLocal(self.id)
    })
  }
`
So, this issue has something related with async data.

**Comment by do-web** [comment]
You could also use something like this:

this.$set(this.items, key, item);
